test/refunder.ts

```diff
@@ -237,36 +239,65 @@ describe("Refunder", function() {
 			await res.wait();
 		});
 
-		it('Successfully refund whitelisted contract function', async () => {
+		it.only('Successfully refund whitelisted contract function', async () => {
 
 			const funcIdAsBytes = generateFuncIdAsBytes('setGreeting(string)');
 			
-			let res = await refunder.whitelistRefundable(greeter.address, funcIdAsBytes, true);
+			let res = await refunder.updateRefundable(greeter.address, funcIdAsBytes, true);
 			await res.wait();
 
-			const balanceBefore = await ethers.provider.getBalance(addr1.address);
+			const testUser = addr1;
+			const balanceBefore = await ethers.provider.getBalance(testUser.address);
 			
 			const text = 'Hello, Tester!';
 			const hexString = strToHex(text);
 			const args = ethers.utils.arrayify(hexString);
 			
-			res = await refunder.connect(addr1).relayAndRefund(greeter.address, funcIdAsBytes, args, {
+			res = await refunder.connect(testUser).relayAndRefund(greeter.address, funcIdAsBytes, args, {
 				gasLimit: 200000
 			});
 
 			let txReceipt = await res.wait();
+			let txCost = res.gasPrice.mul(txReceipt.gasUsed)
+			
+			const balanceAfter = await ethers.provider.getBalance(testUser.address);
+			expect(balanceAfter.lt(balanceBefore), 'Sender was over refunded');
 			
 			expect(txReceipt.events, 'No events are emitted').to.be.ok;
 			expect(txReceipt.events[0].event, 'Invalid event name').to.be.eq('RelayAndRefund');
 
-			const balanceAfter = await ethers.provider.getBalance(addr1.address);
-
 			const difference = balanceBefore.sub(balanceAfter);
```

Comment:
The difference here is in the balances. We need to calculate that we are refunding up to 97% of the TX cost.
Example pseudo-code:

```JavaScript
const cost = res.gasPrice.mul(txReceipt.cumulativeGasUsed)
    const reimbursement = balanceAfter.sub(balanceBefore.sub(cost))
    console.log("TX Cost:", cost.toString())
    console.log("Reimbursement: ", reimbursement.toString())
   ```